### PR TITLE
Updates setup adpat owner doc with cli cmds

### PR DESCRIPTION
**Pull request by iphilpot**
## All Submissions:
-------------------------------------
* [x] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
Only contains setup using portal in doc.

Issue Number: N/A

## What is the new behavior?
-------------------------------------
Adds relevant CLI commands and env vars. All commands tested and verified.

## Does this introduce a breaking change?
-------------------------------------
n0

## Any relevant logs, error output, etc?
-------------------------------------
n/a

## Other information
-------------------------------------
n/a

### CODE REVIEW

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 1]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
 
+> If using the CLI commands please set the following variables:
+
+| Variable | Sample Value | Description |
+|----------|--------------|-------------|
+| `APP_DEVOPS_PROJECT_NAME` | `My Application` | The name of the project representing your application that could include both your Cobalt foundation in addition to any application code, pipelines, boards, or artifacts. |
+| `APPS_DEVOPS_INFRA_REPO_NAME` | `infrastructure` | The name of the repo that will be created in the application Azure DevOps project to host the Cobalt template. |
+| `APP_DEVOPS_INFRA_YML_PATH` | `devops/providers/azure-devops/templates/azure-pipelines.yml` | The path relative to the `APPS_DEVOPS_INFRA_REPO_NAME` root that contains the Cobalt template pipeline to be created for provisioning application resources. |
+| `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `APP_DEVOPS_PROJECT_NAME` resides or will be created. |
+| `GIT_SOURCE_URL` | `https://dev.azure.com/MyOrganization/MyProject/_git/MyCobaltTemplateRepo` | The Git clone URL for the repository hosting the Cobalt template on which your project will be built. |
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
| `COBALT_SOURCE_URL` | `https://github.com/microsoft/cobalt.git` | The Git clone URL for Cobalt (containing all templates including the one to be targeted by this template repository) from which this Cobalt template repository will be sourced. |
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 2]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
   * *Global administrator role* permissions in your Organization's Azure Active Directory tenant to setup service principals
     * If this is not allowed by your organization, step two and the Service Connection creation in step three will need to be completed by someone within your organization with this permission.
 
+> If using the CLI commands please set the following variables:
+
+| Variable | Sample Value | Description |
+|----------|--------------|-------------|
+| `APP_DEVOPS_PROJECT_NAME` | `My Application` | The name of the project representing your application that could include both your Cobalt foundation in addition to any application code, pipelines, boards, or artifacts. |
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
| `TEMPLATE_DEVOPS_PROJECT_NAME` | `My Application` | The name of the project representing your Cobalt template application that serves as your organization's advocated pattern for a specific template. |
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 3]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+| `APPS_DEVOPS_INFRA_REPO_NAME` | `infrastructure` | The name of the repo that will be created in the application Azure DevOps project to host the Cobalt template. |
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
| `TEMPLATE_DEVOPS_INFRA_REPO_NAME` | `az-hello-world` | The name of the repo that will be created in the application Azure DevOps project to host the Cobalt template. |
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 4]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+| `APP_DEVOPS_INFRA_YML_PATH` | `devops/providers/azure-devops/templates/azure-pipelines.yml` | The path relative to the `APPS_DEVOPS_INFRA_REPO_NAME` root that contains the Cobalt template pipeline to be created for provisioning application resources. |
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
| `TEMPLATE_DEVOPS_INFRA_YML_PATH` | `devops/providers/azure-devops/templates/azure-pipelines.yml` | The path relative to the `TEMPLATE_DEVOPS_INFRA_REPO_NAME` root that contains the Cobalt template pipeline to be created for testing and provisioning resources. |
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 5]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+| `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `APP_DEVOPS_PROJECT_NAME` resides or will be created. |
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
| `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `TEMPLATE_DEVOPS_PROJECT_NAME` resides or will be created. |
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 6]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+| `GIT_SOURCE_URL` | `https://dev.azure.com/MyOrganization/MyProject/_git/MyCobaltTemplateRepo` | The Git clone URL for the repository hosting the Cobalt template on which your project will be built. |
+
+Update these values for your environment and application based on the guidance in the table above.
+```bash
+export APP_DEVOPS_PROJECT_NAME=""
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
export TEMPLATE_DEVOPS_PROJECT_NAME=""
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 7]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+export APP_DEVOPS_INFRA_REPO_NAME=""
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
export TEMPLATE_DEVOPS_INFRA_REPO_NAME=""
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 8]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+export APP_DEVOPS_INFRA_YML_PATH=""
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
export TEMPLATE_DEVOPS_INFRA_YML_PATH=""
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 9]
```diff
@@ -27,6 +58,12 @@ This document provides Cobalt users instructions for initializing and integratin
             ![New Project](https://user-images.githubusercontent.com/10041279/63442791-4f868600-c3f9-11e9-91f3-c959654f5a1c.png)
 
         * Select Create
+
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 10]
```diff
@@ -27,6 +58,12 @@ This document provides Cobalt users instructions for initializing and integratin
             ![New Project](https://user-images.githubusercontent.com/10041279/63442791-4f868600-c3f9-11e9-91f3-c959654f5a1c.png)
 
         * Select Create
+    
+    ```
+    az devops configure --defaults organization="$DEFAULT_ORGANIZATION"
+    az devops project create --name "$APP_DEVOPS_PROJECT_NAME" --source-control git --visibility private
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az devops project create --name "$TEMPLATE_DEVOPS_PROJECT_NAME" --source-control git --visibility private
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 11]
```diff
@@ -27,6 +58,12 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    az devops configure -d project="$APP_DEVOPS_PROJECT_NAME"
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az devops configure -d project="$TEMPLATE_DEVOPS_PROJECT_NAME"
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 12]
```diff
@@ -54,6 +91,11 @@ This document provides Cobalt users instructions for initializing and integratin
             | Cobalt-AZ-ISO-Contoso | If the aim is to have a single repository represent a single Cobalt template, and thereafter, to have one repo per template, we recommend a name that reflects the Cobalt Template being deployed. In this naming example, the name assumes this repo will be dedicated to deploying the Cobalt *az-isolated-service-single-region* template |
             | Cobalt-Contoso | If the aim is to use a single repository as ground truth for all potential patterns across your organization, effectively having to manage a combination of Cobalt patterns from a single repo, it's recommended to stick with a name that matches the project name. |
 
+    ```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

        ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 13]
```diff
@@ -54,6 +91,11 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    az repos create --name "$APP_DEVOPS_INFRA_REPO_NAME"
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az repos create --name "$TEMPLATE_DEVOPS_INFRA_REPO_NAME"
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 14]
```diff
@@ -54,6 +91,11 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    az repos import create --git-url https://github.com/microsoft/cobalt.git --repository "$APP_DEVOPS_INFRA_REPO_NAME"
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az repos import create --git-url $COBALT_SOURCE_URL --repository "$TEMPLATE_DEVOPS_INFRA_REPO_NAME"
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 15]
```diff
@@ -83,6 +125,10 @@ This document provides Cobalt users instructions for initializing and integratin
 
             > NOTE: Azure Devops forces a run so expect this to fail. Future steps will resolve this problem.
 
+    ```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion

> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 16]
```diff
@@ -83,6 +125,10 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    az pipelines create --name "$COBALT_PIPELINE_NAME" --repository "$APP_DEVOPS_INFRA_REPO_NAME" --branch master --repository-type tfsgit --yml-path $APP_DEVOPS_INFRA_YML_PATH --skip-run true
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az pipelines create --name "$COBALT_PIPELINE_NAME" --repository "$TEMPLATE_DEVOPS_INFRA_REPO_NAME" --branch master --repository-type tfsgit --yml-path $TEMPLATE_DEVOPS_INFRA_YML_PATH --skip-run true
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 17]
```diff
@@ -142,6 +188,10 @@ This document provides Cobalt users instructions for initializing and integratin
         * Once deployment for storage account is completed, go to the resource and visit the Blobs sub-menu
         * Click [+Container] then create a container name (ex. az-hw-remote-state-container or az-iso-remote-state-container) with private access
 
+```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions. Note this creates the Service Connection within Azure DevOps, but you will still need to setup the Service Principal:

```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 18]
```diff
@@ -142,6 +188,10 @@ This document provides Cobalt users instructions for initializing and integratin
...
+az devops service-endpoint azurerm create --azure-rm-subscription-id $SUBSCRIPTION_ID --azure-rm-subscription-name $SUBSCRIPTION_NAME --azure-rm-tenant-id $TENANT_ID --azure-rm-service-principal-id $SERVICE_PRIN_ID --name $SERVICE_CONN_NAME
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
az devops service-endpoint azurerm create --azure-rm-subscription-id $SUBSCRIPTION_ID --azure-rm-subscription-name "$SUBSCRIPTION_NAME" --azure-rm-tenant-id $TENANT_ID --azure-rm-service-principal-id $SERVICE_PRIN_ID --name "$SERVICE_CONN_NAME"
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 19]
```diff
@@ -188,6 +238,25 @@ This document provides Cobalt users instructions for initializing and integratin
     > Important: Every targeted environment specified within the build pipeline expects a
     > variable group specified with the naming convention `<ENVIRONMENT_NAME> Environment Variables`
 
+    ```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 20]
```diff
@@ -200,6 +269,15 @@ This document provides Cobalt users instructions for initializing and integratin
             | `REMOTE_STATE_ACCOUNT` | `<AZURE_STORAGE_ACCOUNT_NAME>` | The storage container account name created in a previous step that is used to manage the state of this deployment pipeline. The storage Account is shared among all non-prod deployment stages. |
             | `SERVICE_CONNECTION_NAME` | ex. Cobalt Deployment Administrator-`<TenantName>` | The custom name of the service connection configured in a previous Azure Devops step that establishes a connection between the Service Principal and the Azure subscription that it's permissioned for. |
 
+    ```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 21]
```diff
@@ -200,6 +269,15 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    # IMPORTANT: Replace these values as necessary to fit your environment.
+    DEVINT_VAR_GROUP="DevInt $COBALT_VAR_GROUP_ENV_SUFFIX"
+    az pipelines variable-group create --authorize true --name $DEVINT_VAR_GROUP --variables \
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az pipelines variable-group create --authorize true --name "$DEVINT_VAR_GROUP" --variables \
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 22]
```diff
@@ -226,6 +304,24 @@ This document provides Cobalt users instructions for initializing and integratin
 
         * Save the build pipeline
 
+    ```
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
> The following CLI command(s) can be run as an alternative to using the portal-based instructions:

    ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 23]
```diff
@@ -226,6 +304,24 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    az pipelines show --name "Cobalt-Hello-World-Pipeline" -o json > builddef.json
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az pipelines show --name "$COBALT_PIPELINE_NAME" -o json > builddef.json
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 24]
```diff
@@ -226,6 +304,24 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    PIPELINE_ID=$(az pipelines show --name "Cobalt-Hello-World-Pipeline" --query id)
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    PIPELINE_ID=$(az pipelines show --name "$COBALT_PIPELINE_NAME" --query id)
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 25]
```diff
@@ -226,6 +304,24 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    ```
+
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion

Execute the list command to find the Variable Group IDs created earlier. Make note of the IDs as they will need to be added to the build pipeline definition.

    ```bash
    az pipelines variable-group list
    ```
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 26]
```diff
@@ -226,6 +304,24 @@ This document provides Cobalt users instructions for initializing and integratin
...
+    For the workaround, you'll be manually editing the builddef.json file to add the variable group references. At the end of the file, you should see the line `"variableGroups" : null`. Replace the value with the following, replacing the variable group ID placeholders (`0`) with those from the above command for the Infrastructure Pipeline Variables group and DevInt Environment Variables group:
+    ```bash
+    "variableGroups": [
+        { "id": 0 },
+        { "id": 0 }
+    ],
+    ```
+
+    ```
+    az devops invoke --http-method PUT --area build --resource definitions --route-parameters project="TestCobalt" definitionId=10 --in-file builddef.json
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
    az devops invoke --http-method PUT --area build --resource definitions --route-parameters project="$TEMPLATE_DEVOPS_PROJECT_NAME" definitionId=$PIPELINE_ID --in-file builddef.json
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 27]
```diff
@@ -14,6 +14,37 @@ This document provides Cobalt users instructions for initializing and integratin
...
+| `DEFAULT_ORGANIZATION` | `https://dev.azure.com/MyOrganization/` | The full URL path of the organization in which your `APP_DEVOPS_PROJECT_NAME` resides or will be created. |
+| `GIT_SOURCE_URL` | `https://dev.azure.com/MyOrganization/MyProject/_git/MyCobaltTemplateRepo` | The Git clone URL for the repository hosting the Cobalt template on which your project will be built. |
+
+Update these values for your environment and application based on the guidance in the table above.
+```bash
+export APP_DEVOPS_PROJECT_NAME=""
+export APP_DEVOPS_INFRA_REPO_NAME=""
+export APP_DEVOPS_INFRA_YML_PATH=""
+export DEFAULT_ORGANIZATION=""
+export GIT_SOURCE_URL=""
```
Comment (awkwardindustries) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
export COBALT_SOURCE_URL=""
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 28]
```diff
@@ -188,6 +238,25 @@ This document provides Cobalt users instructions for initializing and integratin
...
+        AGENT_POOL='Hosted Ubuntu 1604' \
+        ARM_PROVIDER_STRICT=true \
+        BUILD_ARTIFACT_NAME='drop' \
+        BUILD_ARTIFACT_PATH_ALIAS='artifact' \
+        GO_VERSION='1.12.5' \
+        PIPELINE_ROOT_DIR='devops/providers/azure-devops/templates/infrastructure' \
+        REMOTE_STATE_CONTAINER='BACKENDSTATECONTAINERNAME' \
+        SCRIPTS_DIR='scripts' \
+        TEST_HARNESS_DIR='test-harness/' \
+        TF_DEPLOYMENT_TEMPLATE_ROOT='infra/templates/az-hello-world' \
```
Comment (nmiodice) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]:
```suggestion
        TF_DEPLOYMENT_TEMPLATE_ROOT='infra/templates/$TEMPLATE_DEVOPS_INFRA_REPO_NAME' \
```

File: docs/GETTING_STARTED_ADD_PAT_OWNER.md [hunk 29]
```diff
@@ -188,6 +238,25 @@ This document provides Cobalt users instructions for initializing and integratin
     > Important: Every targeted environment specified within the build pipeline expects a
     > variable group specified with the naming convention `<ENVIRONMENT_NAME> Environment Variables`
 
+    ```
+    # IMPORTANT: Replace these values as necessary to fit your environment.
+    az pipelines variable-group create --authorize true --name "$COBALT_VAR_GROUP_INFRA" --variables \
+        AGENT_POOL='Hosted Ubuntu 1604' \
```
Comment (nmiodice) [inline docs/GETTING_STARTED_ADD_PAT_OWNER.md]: Can we parameterize this to `$AGENT_POOL_NAME`?

**Pull request merged**